### PR TITLE
Testing Spike

### PR DESCRIPTION
**Pull request by plukevdh**
### Are there any dependencies (software or human) that need to be addressed before this PR is merged?

- [x] Add CI testing tasks
- [x] Identifiy and add more tests for edge cases

Test Computer GIF

### What was the problem or feature?
Changes break things, but I didn't know it without tests first.

### What was the solution?
Add tests! 

Taking advantage of the wonderful `rspec-terraform` + `ruby_terraform` gems.

- [x] Security Impact has been considered (if yes please describe)
- [x] Network Impacts have been considered (if yes please describe)

### Where does this work fall on the Good - Fast spectrum?
So fast, so good.

### CODE REVIEW

**Comment by plukevdh** [comment]
Prereq for any future work. This way we won't break as many things moving forward.

File: examples/main.tf [hunk 1]
```diff
@@ -28,7 +28,9 @@ module "missing_one" {
 
   targets = {
     lambda = [
-      "arn:aws:lambda:us-east-1:123456789012:function:summonPatronus"
+      "arn:aws:lambda:us-east-1:123456789012:function:summonPatronus",
+      "arn:aws:lambda:us-east-1:123456789012:function:summonRon",
+      "arn:aws:lambda:us-east-1:123456789012:function:summonHermione"
```
Comment (jedschneider) [inline examples/main.tf:33]: :owl:

File: spec/event_mapping_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  context "resource creation" do
+    let(:event_rules) { @plan.resource_changes_matching(type: "aws_cloudwatch_event_rule") }
+
+    it "creates aws_cloudwatch_event_rules for bus targets" do
+      expect(@plan).to include_resource_creation(type: 'aws_cloudwatch_event_rule').twice
+    end
+
+    it "points to specified bus" do
+      expect(@plan).to include_resource_creation(type: 'aws_cloudwatch_event_rule')
+        .with_attribute_value(:event_pattern, "{\"detail-type\":[\"event.DementorsAppear\"]}")
```
Comment (jedschneider) [inline spec/event_mapping_spec.rb:21]: this is nice that it validates the actual output json (regarding my comment about the `<>` being escaped in highwingio/enrichment-component#15. Lets see how it runs but it might be nice to add some helpers to evaluate this more natively in ruby.

```suggestion
        .with_attribute_value(:event_pattern, {"detail-type" => ["event.DementorAppear"]}.to_json)
```

File: spec/event_mapping_spec.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    end
+
+    context "bus rules" do
+      let(:event_rule) { event_rules.first.change.after }
+
+      it "configures the rule" do
+        expect(event_rule[:name]).to eq("event.DementorsAppear")
+        expect(event_rule[:event_pattern]).to eq({"detail-type": ["event.DementorsAppear"]}.to_json)
```
Comment (jedschneider) [inline spec/event_mapping_spec.rb:29]: oh, there you go. good minds think alike.

**Review by jedschneider — APPROVED** [review]
super interesting. All in.
